app/static/src/router.ts

```diff
@@ -31,7 +34,7 @@ export function initialiseRouter(
         component: appComponent,
         props: {
             initSelectedTab: STATIC_BUILD ? "Options" : undefined
-        } as Parameters<NonNullable<(typeof appComponent)["setup"]>>["0"]
+        } as Parameters<NonNullable<IntersectionAppComponent["setup"]>>["0"]
```

Comment:
this line basically gets the type of the first argument of setup function (which is props), so we can be sure that the selected tab exists for every app type

Reply:
I can see why you've done this but it's quite confusing to read! I don't really love invoking the first parameter of setup as a way of getting out of declaring actual props types - it does save you that maintenance overhead but with something of a loss of clarity! Could we at least declare this type above and put a comment on it?

Reply:
sure yh, sadly i couldnt actually find a way to extract the prop types any other way from defineComponent type which sucks, feel like they couldve made that easier

Reply:
Yeah, agreed. We've usually just defined the types ourselves in the past. It's boring boilerplate, but it's easy to comprehend...